### ISSUE TITLE
Allow toleration customization of Fleet agent

### ISSUE THREAD

**Issue by wbagdon** [rancher/rancher#30471]
**What kind of request is this (question/bug/enhancement/feature request):**
Enhancement

**Steps to reproduce (least amount of steps as possible):**
Create a new cluster with external cloud config

**Result:**
The cluster will be unable to start the fleet agent due to tainted worker nodes `node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule`

**Other details that may be helpful:**
1) Setting the toleration for the Fleet workload allows the Fleet agent to start
2) At this point it is now possible for Fleet to install the vSphere CPI and the taints are lifted from the nodes

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.5.2
- Installation option (single install/HA):
HA

**Comment by jiaqiluo** [comment]
The bug is reproduced on rancher:v2.5.2 single-install 

Steps:
- in Rancher UI, create a custom cluster with `cloud_provider=external`
- register one vSphere node to the cluster
- wait for the cluster to be active 
- go to the `system` project

Results:
The deployment `fleet-agent` is in the error state because it cannot schedule the pod to the node because of the taint. 

<img width="1461" alt="Screen Shot 2021-04-22 at 1 22 52 PM" src="https://user-images.githubusercontent.com/6218999/115780961-a7178a80-a36e-11eb-900a-4b07e338d616.png">

**Comment by jiaqiluo** [comment]
failed to validate the fix on rancher v2.5.8-rc7, fleet version v0.3.5-rc5

The steps are the same as in the previous comment. 

Here are the tolerations on the deployment `fleet-agent`
```
      tolerations:
      - effect: NoSchedule
        key: deployment.Spec.Template.Spec.Tolerations
        operator: Equal
        value: "true"
      - effect: NoSchedule
        key: cattle.io/os
        operator: Equal
        value: linux
```

**Comment by StrongMonkey** [comment]
should be fixed in v0.3.5-rc6.

**Comment by jiaqiluo** [comment]
the fix is validated on rancher v2.5.8-rc7, fleet version v0.3.5-rc6

The `fleet-agent` runs up on the downstream cluster in which the node has the taint `node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule`

The tolerlations on the deployment `fleet-agent`

![image](https://user-images.githubusercontent.com/6218999/115796901-81e34600-a387-11eb-9abe-389f3b495f86.png)

**Comment by atsai1220** [comment]
Running into the issue before this issue. It would make sense to also update the tolerations for `cattle-cluster-agent`.

It does not seem like Fleet will be installed before the cluster checks-in with Rancher but `cattle-cluster-agent` does not have the appropriate tolerations for `node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule`.
